csi/controller_server.go

```diff
@@ -303,6 +298,11 @@ func (cs *ControllerServer) ControllerPublishVolume(ctx context.Context, req *cs
 		}
 	}
 
+	// Check volume frontend settings
+	if existVol.DisableFrontend || existVol.Frontend != string(types.VolumeFrontendBlockDev) {
```

Comment:
There is no need to move condition check `existVol.Frontend != string(types.VolumeFrontendBlockDev)`.

Reply:
In fact, I am fine with moving both since it's weird to check the frontend value if the frontend is disabled.

Reply:
1. The field `vol.DisableFrontend` is decided by the `Attach` request. Hence there is no need to check `existVol.DisableFrontend`.
2. It's better to check `existVol.Frontend != string(types.VolumeFrontendBlockDev)` before the `Attach` call. (There is no need to move this condition check.) If it's false, this function can directly error out rather than continuing call `Attach`.

Reply:
Addressed.